src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -280,7 +281,46 @@ public function compileString($value)
      */
     public static function render($string, $data = [], $deleteCachedView = false)
     {
-        $component = new class($string) extends Component
+        $component = self::temporaryComponent($string);
+        $view = Container::getInstance()
+            ->make(ViewFactory::class)
```

Comment:
Not ideal to have this cyclic dependency.

Reply:
@GrahamCampbell What would you suggest to improve this?